data_pipelines/assets/flood/discharge.py

```diff
@@ -0,0 +1,232 @@
+import os
+from datetime import datetime, timedelta
+import pandas as pd
+
+import xarray as xr
+from dagster import AssetExecutionContext, asset
+import dask.dataframe as dd
+from data_pipelines.settings import settings
+
+from data_pipelines.resources.dask_resource import DaskResource
+from data_pipelines.resources.glofas_resource import CDSClient
+from data_pipelines.utils.flood.config import *
+from data_pipelines.utils.flood.etl.filter_by_upstream import apply_upstream_threshold
+from data_pipelines.utils.flood.etl.raster_converter import RasterConverter
+from data_pipelines.utils.flood.etl.transforms import (
+    compute_flood_intensity,
+    compute_flood_peak_timing,
+    compute_flood_tendency,
+    compute_flood_threshold_percentages,
+    add_geometry,
+)
+from data_pipelines.utils.flood.etl.utils import restrict_dataset_area
+
+from data_pipelines.partitions import discharge_partitions
+
+
+def make_path(*args) -> str:
+    path = os.path.join(*args)
+    os.makedirs(os.path.dirname(path), exist_ok=True)
+    return path
```

Comment:
This doesn't work with a S3 URL, and it creates a local path without raising an error. We will have to rework how the paths are managed, but  for now maybe replace with:
```suggestion
from upath import UPath

def make_path(*args) -> UPath:
    path = UPath(*args)
    path.mkdir(parents=True, exist_ok=True)
    return path
```